dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -563,10 +564,16 @@ private void updateInfo(Workflow oldWorkflow, Workflow newWorkflow) {
         List<Token> tokens = tokenDAO.findZenodoByUserId(user.getId());
         if (!tokens.isEmpty()) {
             Token zenodoToken = tokens.get(0);
-            String refreshUrl = zenodoUrl + "/oauth/token";
-            String payload = "client_id=" + zenodoClientID + "&client_secret=" + zenodoClientSecret
-                    + "&grant_type=refresh_token&refresh_token=" + zenodoToken.getRefreshToken();
-            refreshToken(refreshUrl, zenodoToken, client, tokenDAO, null, null, payload);
+
+            // Check that token is an hour old
+            DateTime updateTime = new DateTime(zenodoToken.getDbUpdateDate());
+            if (updateTime.plusHours(1).isBeforeNow()) {
```

Comment:
You might want to have some buffer time in here, maybe up to a minute. If the token is one second away from expiring here, it's possible that it might expire by the time it is used in an API. We observed some calls taking 10 seconds from the UI, so I generously rounded up to 1 minute as a suggestion -- maybe that's too much.

Reply:
I'll set it to 59 min, I think it should be fine